### ISSUE TITLE
Pre-pipe connection rejection at remote socket causes key error on local socket.

### ISSUE THREAD

**Issue by matt-m-herndon** [codypiersall/pynng#33]
The following exception sometimes occurs when a remote socket rejects a local socket's connection:

```From cffi callback <function _nng_pipe_cb at 0x22E14C90>:
Traceback (most recent call last):
  File "C:\Users\hern6928\Desktop\CPPAR Client\support\cppar-virtual-env\env\lib\site-packages\pynng\nng.py", line 797, in _nng_pipe_cb
    pipe = sock._pipes[pipe_id]
KeyError: 552720521
From cffi callback <function _nng_pipe_cb at 0x22E14C90>:
Traceback (most recent call last):
  File "C:\Users\hern6928\Desktop\CPPAR Client\support\cppar-virtual-env\env\lib\site-packages\pynng\nng.py", line 797, in _nng_pipe_cb
    pipe = sock._pipes[pipe_id]
KeyError: 552720523```

It also appears that "sock._pipes" sometimes doesn't properly capture the rejected status of the connection, as sometimes it still has children when the attempted connections were rejected.

**Comment by codypiersall** [comment]
Ah, my beloved coworker has returned!

This was probably fixed in https://github.com/codypiersall/pynng/commit/e45ab317eb0334eb37cdf2afedb4f8a60e7e9d24; there used to be a race between listen/bind and adding pipe notification events.

I'm hoping to get a new release out whenever I finish updating [the docs](https://pynng.readthedocs.io).

**Comment by codypiersall** [comment]
Now that 0.4.0 is released, I'm going to close this, since I'm pretty sure it was was fixed in e45ab31.  Please reopen it if this issue still occurs!
